docs/api.md

```diff
@@ -0,0 +1,151 @@
+# API Documentation
+
+## High Level Concepts
+
+The *Network Isolator Module* is common code for all implementations of
+IP-per-container networking in Mesos.  It speaks the `isolator` module API with
+Mesos and calls plug-ins that do the actual network implementation.
+
+We define two plug-in APIs:
+
+ 1. IP Address Management (IPAM)
+ 2. Network Virtualizer
+
+This allows these aspects of network management to be separately customized.
+
+A *netgroup* is the name given to a set of logically-related IPs that are
+allowed to communicate within themselves. For example, one might want to create
+separate netgroups for `dev`, `testing`, `qa` and `prod` deployment
+environments.
+
+## Plug-Ins
+
+Plug-ins are invoked as binary executables.  The caller then passes a JSON
+blob containing the request data over `stdin` and the plugin responds by
+writing a response JSON blob to `stdout`.
+
+Calls to plug-ins are blocking.
+
+A new instance of the plug-in executable is launched for each request. (In the
+future we may pipeline requests to avoid the overhead of starting a new
+process.)
+
+
+## IPAM Plug-In API
+
+The IPAM plug-in ensures that containers receive unique IP addresses.  The
+Network Isolator Module `allocate`s IPs when it needs them for new containers,
```

Comment:
`allocate`s and `release`s look weird in the github markdown view. Maybe try: The Network Isolator Module will `allocate` IPs when it needs them for new containers, and `release` them when it deems it has too many. ?